.github/workflows/e2e-test.yml

```diff
@@ -6,85 +6,38 @@ on:
       - master
 
 jobs:
-  ubuntu-16:
-    runs-on: ubuntu-16.04
-    name: 'Ubuntu 16.04'
+  e2e:
+    runs-on: ${{ matrix.os }}
+    name: ${{ matrix.os }}
+    strategy:
+      fail-fast: true
+      matrix:
+        os: [ubuntu-16.04, ubuntu-18.04, ubuntu-20.04]
+        password: [root]
+        node: [12.x]
```

Comment:
The cool thing about matricies is we can use some keys as variables - in this case, when updating node, we just have to look for the node version here instead of in the setup node action